nesta/core/orms/nih_orm.py

```diff
@@ -2,116 +2,122 @@
 NIH schema
 ==============
 
-The schema for the World RePORTER data. Note that the schema
-was automatically generated based on an IPython hack session
-using the limits and properties of the data from the 
-World ExPORTER (which explains the unusual VARCHAR limits).
+The schema for the World RePORTER data.
 '''
 
 from sqlalchemy.ext.declarative import declarative_base
-from sqlalchemy.dialects.mysql import VARCHAR, TEXT
-from sqlalchemy.types import INTEGER
+from sqlalchemy.dialects.mysql import VARCHAR as _VARCHAR
+from sqlalchemy.dialects.mysql import TEXT as _TEXT
+from sqlalchemy.types import INTEGER, JSON, DATETIME
 from sqlalchemy import Column, Table
 
 
 Base = declarative_base()
+TEXT = _TEXT(collation='utf8mb4_unicode_ci')
+VARCHAR = lambda *args, **kwargs: _VARCHAR(*args, 
+                                           collation='utf8mb4_unicode_ci', 
+                                           **kwargs)
+
 
 class Projects(Base):
     __tablename__ = 'nih_projects'
 
-    application_id = Column(INTEGER, primary_key=True)
+    application_id = Column(INTEGER, primary_key=True, autoincrement=False)
     activity = Column(VARCHAR(3))
     administering_ic = Column(VARCHAR(2))
     application_type = Column(INTEGER)
     arra_funded = Column(VARCHAR(1))
-    award_notice_date = Column(VARCHAR(19))
-    budget_start = Column(VARCHAR(10))
-    budget_end = Column(VARCHAR(10))
-    cfda_code = Column(VARCHAR(23))
-    core_project_num = Column(VARCHAR(32), index=True)
-    ed_inst_type = Column(VARCHAR(70))
-    foa_number = Column(VARCHAR(14))
-    full_project_num = Column(VARCHAR(35))
-    funding_ics = Column(VARCHAR(291))
-    funding_mechanism = Column(VARCHAR(23))
-    fy = Column(INTEGER)
-    ic_name = Column(VARCHAR(79))
-    org_city = Column(VARCHAR(26))
-    org_country = Column(VARCHAR(14))
-    org_dept = Column(VARCHAR(30))
+    award_notice_date = Column(DATETIME)
+    budget_start = Column(DATETIME)
+    budget_end = Column(DATETIME)
+    cfda_code = Column(TEXT)
+    core_project_num = Column(VARCHAR(50), index=True)
+    ed_inst_type = Column(TEXT)
+    foa_number = Column(TEXT)
+    full_project_num = Column(VARCHAR(50), index=True)
+    funding_ics = Column(JSON)
+    funding_mechanism = Column(TEXT)
+    fy = Column(INTEGER, index=True)
+    ic_name = Column(VARCHAR(100), index=True)
+    org_city = Column(VARCHAR(50), index=True)
+    org_country = Column(VARCHAR(50), index=True)
+    org_dept = Column(VARCHAR(100), index=True)
     org_district = Column(INTEGER)
-    org_duns = Column(VARCHAR(75))
-    org_fips = Column(VARCHAR(2))
+    org_duns = Column(JSON)
+    org_fips = Column(VARCHAR(2), index=True)
     org_ipf_code = Column(INTEGER)
-    org_name = Column(VARCHAR(92))
-    org_state = Column(VARCHAR(2))
+    org_name = Column(VARCHAR(100), index=True)
+    org_state = Column(VARCHAR(2), index=True)
```

Comment:
> performance implications

Takes up a little more disk (which is cheap), but since we have 2 million rows it will lead to a factor of 300k faster searching and joining. Since I'm writing some data-getters at the end of all this, figured that it would be useful!

Reply:
> When to use VARCHAR vs TEXT?

Sorry, was going to comment on before changing from 'Draft' PR... Generally, I'm moving towards only using `VARCHAR` if the field needs to be indexed, otherwise use `TEXT` since it means we don't need to manage arbitrary `VARCHAR` sizes that are difficult to read. The only exception would be for short codes, where I think the `n` of `VARCHAR(n)` can give a little clarity (it would be worth indexing such an example anyway).